rai/lib/numbers.cpp

```diff
@@ -443,13 +442,13 @@ rai::uint512_union rai::sign_message (rai::raw_key const & private_key, rai::pub
 	return result;
 }
 
-void rai::deterministic_key (rai::uint256_union const & seed_a, uint32_t index_a, rai::uint256_union & prv_a)
+void rai::deterministic_key (rai::uint256_union const & seed_a, uint64_t index_a, rai::uint256_union & prv_a)
 {
 	blake2b_state hash;
 	blake2b_init (&hash, prv_a.bytes.size ());
 	blake2b_update (&hash, seed_a.bytes.data (), seed_a.bytes.size ());
 	rai::uint256_union index (index_a);
-	blake2b_update (&hash, reinterpret_cast<uint8_t *> (&index.dwords[7]), sizeof (uint32_t));
+	blake2b_update (&hash, reinterpret_cast<uint8_t *> (&index.dwords[7]), sizeof (index.dwords[7]));
```

Comment:
Why convert index to a `rai::uint256_t`? Are we changing the endianness? If so, I think we should use boost's functions explicitly for that.